components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1052,9 +1074,96 @@ void ConversationDriver::PerformAssistantGeneration(
   auto data_completed_callback =
       base::BindOnce(&ConversationDriver::OnEngineCompletionComplete,
                      weak_ptr_factory_.GetWeakPtr(), current_navigation_id);
-  engine_->GenerateAssistantResponse(is_video, page_content, chat_history_,
-                                     input, std::move(data_received_callback),
-                                     std::move(data_completed_callback));
+  bool should_refine_page_content =
+      page_content.length() > GetCurrentModel()
+                                  .options->get_leo_model_options()
+                                  ->max_page_content_length &&
+      input != l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_SUMMARIZE_PAGE) &&
+      features::IsPageContentRefineEnabled();
+  if (!text_embedder_ && should_refine_page_content) {
+    if (universal_qa_model_path_.empty()) {
+      universal_qa_model_path_ =
+          leo_local_models_updater_->GetUniversalQAModel();
+    }
+    // Tasks in TextEmbedder are run on |embedder_task_runner|. The
+    // text_embedder_ must be deleted on that sequence to guarantee that pending
+    // tasks can safely be executed.
+    scoped_refptr<base::SequencedTaskRunner> embedder_task_runner =
+        base::ThreadPool::CreateSequencedTaskRunner(
+            {base::MayBlock(), base::TaskPriority::BEST_EFFORT});
```

Comment:
`BEST_EFFORT` is not the right choice, such tasks are considered background and may be run with a notable delay (many seconds). Your use case here is `USER_VISIBLE` or `USER_BLOCKING`.

Reply:
fixed in 6ccf528315ebf8400be0bf904c07f7ec4b0a58d4